src/spanningtrees/prim.jl

```diff
@@ -1,60 +1,84 @@
 immutable PrimHeapEntry{T<:Real}
+    vertex::Int
     edge::Edge
     dist::T
 end
 
 isless(e1::PrimHeapEntry, e2::PrimHeapEntry) = e1.dist < e2.dist
 
-"""Performs [Prim's algorithm](https://en.wikipedia.org/wiki/Prim%27s_algorithm)
+"""
+Checks whether a given vertex `v` is a member of a Vector{PrimHeapEntry}
+"""
+function contains{T<:Real}(pq::Vector{PrimHeapEntry{T}}, v::Int) :: Bool
+    for item in pq
+        v == item.vertex && return true
+    end
+    return false
+end
+
+"""
+Takes a Vector{PrimHeapEntry} `pq` and a vertex `v` as arguments and returns the
+dist key of the vertex.
+"""
+function get_dist{T<:Real}(pq::Vector{PrimHeapEntry{T}}, v::Int) :: T
+    for item in pq
+        v == item.vertex && return item.dist
```

Comment:
In that case, since you're calling `contains` anyway, can you merge `contains` and `get_dist` to return, say, `(true, item.dist)` or `(false, 0)` and just make one O(n) call?